### PR TITLE
Adds step_until! to integrate solution up to prescribed time

### LINKED ISSUE #182: Compute solution at time `t` that is not necessarily an integer multiple of timestep `dt`
Currently FourierFlows.jl can only find solutions at integer multiples of the prescribed timestep `dt`.

It would be useful to be able to compute the solution at any time `t` even if that is not necessarily an integer multiple of `dt`.

One idea to do that is:

prescribe `t` and `dt`:
- `nstep = Int(ceil(t/dt))`; this way `(nstep-1)*dt < t <= nstep*dt`.
- compute solution for up to `nstep*dt`
- output a linear interpolation of the solution between the last and the previous-to-last timesteps, e.g.,
```julia
(nstep*dt - t) / dt * sol[nstep-1] + (t-(nstep-1)*dt) / dt * sol[nstep]
```

In the case that `t=nstep*dt` then the above becomes simply `sol[nstep]` (similarly for if `t=(nstep-1)*dt`).

[DifferentialEquations.jl](https://github.com/SciML/DifferentialEquations.jl) does interpolate the output solution to the prescribed time `t`; I'm not sure what exactly the do there.

[This feature request came after discussions I had with @rveltz]

### PR DESCRIPTION
**Pull request by navidcy**
Closes #182.

### CODE REVIEW

File: src/timesteppers.jl [hunk 1]
```diff
@@ -480,3 +480,30 @@ function getetdcoeffs(dt, L; ncirc=32, rcirc=1)
...
+  return nothing
+end
+
+function step_until!(prob::Problem, ::Union{ETDRK4TimeStepper, FilteredETDRK4TimeStepper}, stop_time)
+  @error "step_until! requires fully explicit time stepper; does not work with ETDRK4"
+  return nothing
+end
+
+function step_until!(prob::Problem, timestepper, stop_time)
+  Δt = stop_time - prob.clock.t
```
Comment (navidcy) [inline src/timesteppers.jl:500]: perhaps we should make sure that Δt>0?

Comment (glwagner) [reply src/timesteppers.jl:500]: Check that `stop_time` is greater than `prob.clock.t`.

Comment (navidcy) [reply src/timesteppers.jl:500]: How to do that without if?

**Comment by navidcy** [comment]
The check 
```
Δt = stop_time > prob.clock.t ? stop_time - prob.clock.t : error("stop_time must be > prob.clock.t")
```
leads to 2-3% slowdown. I guess that's OK. I'm merging.

**Pull request merged**